deepinterpolation/cli/training.py

```diff
@@ -0,0 +1,97 @@
+import argschema
+import json
+from pathlib import Path
+
+from deepinterpolation.cli.schemas import TrainingInputSchema
+from deepinterpolation.generic import ClassLoader
+
+
+class Training(argschema.ArgSchemaParser):
+    default_schema = TrainingInputSchema
+
+    def run(self):
+        self.logger.name = type(self).__name__
+
+        outdir = Path(self.args['training_params']['output_path'])
+        if self.args["output_full_args"]:
+            full_args_path = outdir / "training_full_args.json"
+            with open(full_args_path, "w") as f:
+                json.dump(self.args, f, indent=2)
+            self.logger.info(f"wrote {full_args_path}")
+        uid = self.args['run_uid']
+
+        # We create the output model filename if empty
+        if self.args['training_params']["model_string"] == "":
+            self.args['training_params']["model_string"] = (
+                self.args["network_params"]["name"] + "_" +
+                self.args["training_params"]["loss"]
+            )
+
+        # We pass on the uid
+        self.args["training_params"]["run_uid"] = uid
+
+        # We pass on folder
+        self.args["training_params"]["output_dir"] = \
+            self.args['training_params']['output_path']
+
+        # We convert to old schema
+        self.args['training_params']['nb_gpus'] = 2 * \
+            int(self.args['training_params']['multi_gpus'])
+
+        # Forward parameters to the training agent
+        self.args['training_params']['steps_per_epoch'] = \
+            self.args['generator_params']['steps_per_epoch']
+        self.args['training_params']['batch_size'] = \
+            self.args['generator_params']['batch_size']
+
+        # save the json parameters to 2 different files
+        training_json_path = outdir / f"{uid}_training.json"
+        with open(training_json_path, "w") as f:
+            json.dump(self.args['training_params'], f,  indent=2)
+        self.logger.info(f"wrote {training_json_path}")
+
+        generator_json_path = outdir / f"{uid}_generator.json"
+        with open(generator_json_path, "w") as f:
+            json.dump(self.args['generator_params'], f, indent=2)
+        self.logger.info(f"wrote {generator_json_path}")
+
+        network_json_path = outdir / f"{uid}_network.json"
+        with open(network_json_path, "w") as f:
+            json.dump(self.args['network_params'], f, indent=2)
+        self.logger.info(f"wrote {network_json_path}")
+
+        test_generator_json_path = outdir / f"{uid}_test_generator.json"
+        with open(test_generator_json_path, "w") as f:
+            json.dump(self.args['test_generator_params'], f, indent=2)
+        self.logger.info(f"wrote {test_generator_json_path}")
+
+        generator_obj = ClassLoader(generator_json_path)
+        data_generator = generator_obj.find_and_build()(generator_json_path)
```

Comment:
I gather that this method reads its parameters from that JSON file. I feel like it would be cleaner if the method accepted a dict of args. You could then move the JSON-writing code into `generator_obj.find_and_build()`. In that case, it's not that you're writing the args you want to use to a file and reading them back in, but you passing args to `generator_obj.find_and_build()` and `find_and_build` is writing out its configuration to a log file.

That feels to me like an arrangement that is less likely to read in the wrong parameters by accident (if, for instance, someone ever changes how this JSON file gets written and forgets to change how `generator_obj.find_and_build()` gets called.